core-addon/Core.js

```diff
@@ -194,6 +194,11 @@ module.exports = class Core {
         return this._unenrollStudy(message.data.studyID);
       case "unenrollment":
         return this._unenroll();
+      case "uninstall-rally":
+        // If a user is uninstalling Rally because they do not qualify to be 
+        // on the platform (e.g. they are non-US), uninstall it for them on request.
+        browser.management.uninstallSelf({ showConfirmDialog: false });
+        return Promise.resolve();
```

Comment:
```suggestion
        return browser.management.uninstallSelf({ showConfirmDialog: false });
```
That API returns a promise, we should return it back to the caller.

Reply:
So I had this originally, but it caused the unit test I added to timeout. Any ideas there?

Reply:
You probably need [this line](https://github.com/mozilla-rally/rally-core-addon/blob/3a145ca14bc141eb5bb82898e7508c04ecf61aeb/tests/core-addon/unit/Core.test.js#L222). I'd recommend going with what I suggested you in the other comment, it will save you from the need of writing the unit test at all :D

Reply:
Yeah, I think the test already has that line but still times out. It's a problem with the `browser.management.uninstallSelf` mock itself. Maybe they've made it so it really simulates what it does & never actually resolves! At any rate, if `updatePlatformEnrollment(false)` can be overloaded to just follow the uninstall path, that works for me too!